CMakeLists.txt

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (C) 2024 Renesas Electronics Corporation.
+# Copyright (C) 2024 EPAM Systems, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+cmake_minimum_required(VERSION 3.19)
+
+# ##############################################################################
+# Generate gRPC stubs
+# ##############################################################################
+
+project(aos_core_api)
+
+# ##############################################################################
+# Options
+# ##############################################################################
+
+option(GENERATE_CORE_API_IAM "build core api iam lib" OFF)
+option(GENERATE_CORE_API_SM "build core api sm lib" OFF)
+
+message(STATUS)
+message(STATUS "${CMAKE_PROJECT_NAME} configuration:")
+message(STATUS "GENERATE_CORE_API_IAM        = ${GENERATE_CORE_API_IAM}")
+message(STATUS "GENERATE_CORE_API_SM         = ${GENERATE_CORE_API_SM}")
+
+# ##############################################################################
+# Compiler flags
+# ##############################################################################
+
+add_compile_options(-fPIC -Wall -Werror -Wextra -Wpedantic)
+set(CMAKE_CXX_STANDARD 17)
+
+# ##############################################################################
+# Dependencies
+# ##############################################################################
+
+find_package(gRPC CONFIG REQUIRED)
+find_package(Protobuf REQUIRED)
+
+set(PROTO_DST_DIR "${CMAKE_CURRENT_BINARY_DIR}/gen")
+set(PROTO_SRC_DIR "${CMAKE_CURRENT_LIST_DIR}/proto")
+set(GENERATE_EXTENSIONS ".grpc.pb.h" ".grpc.pb.cc")
+
+file(MAKE_DIRECTORY ${PROTO_DST_DIR})
+
+if(CMAKE_CROSSCOMPILING)
+    find_program(_GRPC_CPP_PLUGIN_EXECUTABLE grpc_cpp_plugin)
+else()
+    set(_GRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:gRPC::grpc_cpp_plugin>)
+endif()
+
+# ##############################################################################
+# Targets
+# ##############################################################################
+
+add_subdirectory(proto)
```

Comment:
if we move all targets to api repo so than maybe we need to add here target for protocol 

https://github.com/aosedge/aos_core_mp_cpp/blob/8c7ec0f23cd8129903c0573a89fc638d53c03acb/external/api/CMakeLists.txt#L11

Reply:
done